ui/gl/gl_context_glfw.cc

```diff
@@ -0,0 +1,174 @@
+    default:
+      NOTREACHED();
+      return NULL;
+  }
+}
+
+
+bool GLContextGlfw::Initialize(
+    GLSurface* compatible_surface, GpuPreference gpu_preference) {
+  DCHECK(compatible_surface);
+  DCHECK(!surface_);
+  DCHECK(!context_);
+
+  surface_ = compatible_surface;
+  context_ = reinterpret_cast<gfx::AcceleratedWidget>(surface_->GetHandle());
+
+  return true;
+}
+
+void GLContextGlfw::Destroy() {
+  if (surface_) {
+    surface_ = NULL;
+  }
+  if (context_) {
+    context_ = NULL;
+  }
+}
+
+bool GLContextGlfw::MakeCurrent(GLSurface* surface) {
+  DCHECK(surface_);
+  if (IsCurrent(surface))
+      return true;
+
+  ScopedReleaseCurrent release_current;
+  TRACE_EVENT2("gpu", "GLContextEGL::MakeCurrent",
+               "context", surface_,
+               "surface", surface);
+
+  if (unbind_fbo_on_makecurrent_ && glfwGetCurrentContext() != NULL) {
+    glBindFramebufferEXT(GL_FRAMEBUFFER, 0);
+  }
+
+  // Make the surface's context current.
+  glfwMakeContextCurrent(context_);
+
+  // Set this as soon as the context is current, since we might call into GL.
+  SetRealGLApi();
+
+  SetCurrent(surface);
+  if (!InitializeDynamicBindings()) {
+    return false;
+  }
+
+  if (!surface->OnMakeCurrent(this)) {
+    LOG(ERROR) << "Could not make current.";
+    return false;
+  }
+
+  surface->OnSetSwapInterval(swap_interval_);
+
+  release_current.Cancel();
+  return true;
+}
+
+void GLContextGlfw::SetUnbindFboOnMakeCurrent() {
+  unbind_fbo_on_makecurrent_ = true;
+}
+
+void GLContextGlfw::ReleaseCurrent(GLSurface* surface) {
+  if (!IsCurrent(surface))
+    return;
+
+  if (unbind_fbo_on_makecurrent_)
+    glBindFramebufferEXT(GL_FRAMEBUFFER, 0);
+
+  SetCurrent(NULL);
+  glfwMakeContextCurrent(NULL);
+}
+
+bool GLContextGlfw::IsCurrent(GLSurface* surface) {
+  DCHECK(surface_);
+  DCHECK(context_);
+
+  bool native_context_is_current = context_ == glfwGetCurrentContext();
+
+  // If our context is current then our notion of which GLContext is
+  // current must be correct. On the other hand, third-party code
+  // using OpenGL might change the current context.
+  DCHECK(!native_context_is_current || (GetRealCurrent() == this));
+
+  if (!native_context_is_current)
+    return false;
+
+  return true;
+}
+
+void* GLContextGlfw::GetHandle() {
+  return (void *)context_;
+}
+
+void GLContextGlfw::OnSetSwapInterval(int interval) {
+  DCHECK(IsCurrent(NULL) && GLSurface::GetCurrent());
+
+  glfwSwapInterval(interval);
+  swap_interval_ = interval;
+  GLSurface::GetCurrent()->OnSetSwapInterval(interval);
+}
+
+std::string GLContextGlfw::GetExtensions() {
+  return GLContext::GetExtensions();
+}
+
+bool GLContextGlfw::WasAllocatedUsingRobustnessExtension() {
+  int robust = glfwGetWindowAttrib(context_, GLFW_CONTEXT_ROBUSTNESS);
+  return (robust != GLFW_NO_ROBUSTNESS);
+}
+
+GLContextGlfw::~GLContextGlfw() {
+  Destroy();
+}
+
+#if !defined(OS_ANDROID)
+bool GLContextGlfw::GetTotalGpuMemory(size_t* bytes) {
+  DCHECK(bytes);
+  *bytes = 0;
```

Comment:
This seems like maybe it's unimplemented, not a big deal but might leave a comment if so.